init/devuan/walinuxagent

```diff
@@ -0,0 +1,354 @@
+	fi
+	echo "${PID}"
+}
+
+recheck_status() {
+	# after an attempt to stop the daemon, re-check the status
+	# and take any further actions required.
+	# (NB: at the moment, we only re-check once. Possible improvement
+	# would be to iterate the re-check up to a given maximum tries). 
+	local STATUS NEWSTATUS
+	get_status
+	STATUS=$?
+	debugmsg "stop_waagent: status is now ${STATUS}"
+	# ideal if stop has been successful: STATUS=1 - no daemon process
+	case ${STATUS} in 
+		0)
+			# stop didn't work
+			# what to do? maybe try kill -9 ? 
+			debugmsg "recheck_status: ERROR: unable to stop waagent"
+			debugmsg "recheck_status: trying again with kill -9"
+			kill_daemon_from_pid 1
+			# probably need to check status again? 
+			get_status
+			NEW_STATUS=$?
+			if [ "x${NEW_STATUS}" == "x1" ] ; then
+				debugmsg "recheck_status: successfully stopped."
+				log_end_msg 0 || true
+			else
+				# could probably do something more productive here
+				debugmsg "recheck_status: unable to stop daemon - giving up"
+				log_end_msg 1 || true
+				exit 1
+			fi
+		;;
+		1) 
+			# THIS IS THE EXPECTED CASE: daemon is no longer running and
+			debugmsg "recheck_status: waagent daemon stopped successfully."
+			log_end_msg 0 || true
+		;;
+		2) 
+			# so weird that we can't figure out what's going on							
+			debugmsg "recheck_status: ERROR: unable to determine waagent status"
+			debugmsg "recheck_status: manual intervention required"
+			log_end_msg 1 || true
+			exit 1
+		;;
+	esac
+}
+
+start_waagent() {
+	# we use start-stop-daemon for starting waagent
+	local STATUS
+	get_status
+	STATUS=$?
+	# check the status value - take appropriate action
+	debugmsg "start_waagent: STATUS=${STATUS}"
+	case "${STATUS}" in 
+		0) 
+			debugmsg "start_waagent: waagent is already running"
+			log_daemon_msg "waagent is already running"
+			log_end_msg 0 || true
+		;;
+		1)
+			# not running (we ignore presence/absence of pidfile)
+			# just start waagent
+			debugmsg "start_waagent: waagent is not currently running"
+			log_daemon_msg "Starting ${NAME} daemon"
+#			start-stop-daemon --start --quiet --pidfile ${PIDFILE} --background  --name "${NAME}" --exec ${INTERPRETER} -- ${DAEMON} ${DAEMON_ARGS}
+# we don't create a pidfile now
+			start-stop-daemon --start --quiet --background  --name "${NAME}" --exec ${INTERPRETER} -- ${DAEMON} ${DAEMON_ARGS}
+			log_end_msg $? || true
+		;;
+		2)
+			# get_status can't figure out what's going on.
+			# try doing a stop to clean up any mess, then attempt to start waagent
+			# will probably require manual intervention
+			# (REVISIT: in this case, maybe we should re-check the status?)
+			debugmsg "start_waagent: unable to determine current status"
+			debugmsg "start_waagent: trying to stop waagent first, and then start it"
+			stop_waagent
+			log_daemon_msg "Starting ${NAME} daemon"
+#			start-stop-daemon --start --quiet --pidfile ${PIDFILE} --background --name ${NAME} --exec ${INTERPRETER} -- ${DAEMON} ${DAEMON_ARGS}
```

Comment:
leftover?

Reply:
I've now removed the commented-out codelines - I'd retained them for reference during local testing